lib/types.h

```diff
@@ -105,9 +79,9 @@ template <class Archive> void serialize(Archive& archive, Color& c) {
  */
 struct Box {
     float surface_area() const {
-        auto e0 = max[Axis::X] - min[Axis::X];
-        auto e1 = max[Axis::Y] - min[Axis::Y];
-        auto e2 = max[Axis::Z] - min[Axis::Z];
+        auto e0 = max[0] - min[0];
```

Comment:
The idea that constants should be also namespaced and should not be global. This place can be actually reverted, I changed it before I introduced the operator [] working with Axis.